satellite/satellitedb/overlaycache.go

```diff
@@ -902,6 +902,50 @@ func (cache *overlaycache) UpdateUptime(ctx context.Context, nodeID storj.NodeID
 	return getNodeStats(dbNode), Error.Wrap(tx.Commit())
 }
 
+// AllPieceCounts returns a map of node IDs to piece counts from the db.
+func (cache *overlaycache) AllPieceCounts(ctx context.Context) (_ map[storj.NodeID]int, err error) {
+	defer mon.Task()(&ctx, err)
+
+	// NB: `All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number` selects node
+	// ID and piece count from the nodes table where piece count is not zero.
+	rows, err := cache.db.All_Node_Id_Node_PieceCount_By_PieceCount_Not_Number(ctx)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+
+	pieceCounts := make(map[storj.NodeID]int)
+
+	var nodeID storj.NodeID
+	for _, row := range rows {
+		// NB: assumes node IDs in nodes table are always valid.
+		copy(nodeID[:], row.Id)
```

Comment:
I read the comment, but I'm on the fence.  Isn't `NodeIDFromBytes` clearer and more robust if we change implementation?

Reply:
Ah! Nice, I forgot about that, thanks!